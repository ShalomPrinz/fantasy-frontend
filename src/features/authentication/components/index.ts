import AuthForm from "./AuthForm";

export { AuthForm };
export type { AuthProps } from "./AuthForm";
