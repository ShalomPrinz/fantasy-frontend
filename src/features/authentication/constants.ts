/* API Endpoints */

export const registerUrl = "/register";
export const userInfoUrl = "/user";
