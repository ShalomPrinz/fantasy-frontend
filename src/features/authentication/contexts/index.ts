export { UserProvider, UserState, useUser } from "./UserContext";
