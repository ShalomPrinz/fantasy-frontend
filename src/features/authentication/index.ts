export { UserProvider, UserState, useUser } from "./contexts";
export { Login, Register } from "./pages";
export { loadIdToken } from "./services";
