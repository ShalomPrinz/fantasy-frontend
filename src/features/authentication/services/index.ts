export { saveIdToken, loadIdToken, removeIdToken } from "./storage";
export { getUserInfo, loginUser, logoutUser, registerUser } from "./api";
