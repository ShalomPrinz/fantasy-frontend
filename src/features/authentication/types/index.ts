export { parseUser } from "./User";
export type { LoginUser, RegisterUser } from "./User";
