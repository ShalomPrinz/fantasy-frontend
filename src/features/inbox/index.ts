export { Inbox } from "./pages";
export { Message } from "./types";
