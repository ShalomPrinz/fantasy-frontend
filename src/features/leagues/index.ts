export { CreateLeague, InviteMembers, League, Leagues } from "./pages";
export type { LeagueInfo } from "./types";
