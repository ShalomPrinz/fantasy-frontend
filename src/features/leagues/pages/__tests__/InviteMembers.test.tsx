import { UserState } from "features/authentication";
import {
  mockLeagueUndefined,
  mockUser,
  renderEffectsSleep,
  renderWithRouterAndUser,
} from "setupTests";

import InviteMembers from "../InviteMembers";

const renderComponent = (path?: string) =>
  renderWithRouterAndUser(<InviteMembers />, path);

const renderComponentSleep = async (path?: string) =>
  await renderEffectsSleep(<InviteMembers />, false, path);

describe("InviteMembers", () => {
  it("should render InviteMembers component if loading user", async () => {
    mockUser({ state: UserState.LOADING_USER });
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render InviteMembers component if no user is logged in", async () => {
    mockUser({ state: UserState.NO_LOGGED_USER });
    const { asFragment } = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  });

  describe("If User is logged in", () => {
    it("should render InviteMembers component if user is not a member of the league", async () => {
      mockUser({ state: UserState.LOGGED_USER });
      const { asFragment } = await renderComponentSleep();
      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("Runtime Errors", () => {
    it("should cause uncaught error and render Error Boundary", () => {
      mockUser({ state: UserState.LOGGED_USER });
      mockLeagueUndefined();

      const { asFragment } = renderComponent();
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
