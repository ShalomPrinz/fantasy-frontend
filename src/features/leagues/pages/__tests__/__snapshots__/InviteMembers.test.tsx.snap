// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteMembers If User is logged in should render InviteMembers component if user is not a member of the league 1`] = `
<DocumentFragment>
  <h1
    class="text-center my-5 p-5 bg-danger text-white w-75 mx-auto"
  >
    You are not member of this league.
  </h1>
</DocumentFragment>
`;

exports[`InviteMembers Runtime Errors should cause uncaught error and render Error Boundary 1`] = `
<DocumentFragment>
  <div
    class="text-center my-5 p-4 bg-danger text-white w-75 mx-auto"
  >
    <h2>
      Invite Members Page Not Available
    </h2>
    <button
      class="fs-4 bg-white rounded"
      type="button"
    >
      Refresh
    </button>
  </div>
</DocumentFragment>
`;

exports[`InviteMembers should render InviteMembers component if loading user 1`] = `
<DocumentFragment>
  <h1
    class="text-center my-5 p-5 bg-info text-white w-75 mx-auto"
  >
    Loading...
  </h1>
</DocumentFragment>
`;

exports[`InviteMembers should render InviteMembers component if no user is logged in 1`] = `
<DocumentFragment>
  <h1
    class="text-center my-5 p-5 bg-danger text-white w-75 mx-auto"
  >
    Please Log In to invite members to your league
  </h1>
</DocumentFragment>
`;
